### DISCUSSION TITLE
Issue logging into account -- chroot(/run/dovecot/empty) failed: Operation not permitted

### DISCUSSION THREAD

**Discussion by benniekiss** [docker-mailserver/docker-mailserver#3172] [General]
Hello,

I had a working configuration of the mailserver running for sometime, then about a week ago I was no longer able to login and the idle CPU usage increased. I'm running the edge ghcr container through podman.

When I attempt to login, I receive these errors in the logs:

`dovecot anvil: Fatal: chroot(/run/dovecot/empty) failed: Operation not permitted`
`dovecot Error: service(anvil): command startup failed, throttling for 4.000 secs`
`dovecot auth: Error: read(anvil-auth-penalty) failed: read(size=1024) failed: Connection reset by peer`
`dovecot: auth: Error: net_connect_unix(anvil-auth-penalty) failed: Permission denied`
`dovecot: imap-login: Fatal: chroot(/run/dovecot/login) failed: Operation not permitted`
`dovecot: master: Error: service(imap-login): command startup failed, throttling for 2.000 secs`

I haven't been able to figure the issue out myself and was wondering if anyone had advice. The issue continued whether I started the mailserver from scratch or from a backup

**Comment by polarathene** [comment]
If you didn't run any updates, check that you're not low on disk space (be mindful of partitions) or memory. If you did perform updates, you should be more explicit about what that was.

**Comment by benniekiss** [comment]
It actually seems directly related to this issue 
https://github.com/docker-mailserver/docker-mailserver/issues/3110#issuecomment-1445313587

I run dms as a user (rootless) podman container. I had been running it for probably a month or two with no issues on the edge branch. So it could be podman updated and changed things. 

I plan to investigate the proposed solutions in that issue, but it doesn’t seem like people had success.

**Reply by polarathene** [reply]
> I plan to investigate the proposed solutions in that issue, but it doesn’t seem like people had success.

We resolved that as best we could for v12. It should ideally have improved support for podman rootless, but maybe not?

It does look like Dovecot is configured to run with `chroot` atm, I don't think we need that, so if you still have problems and look into how to disable it and find that works, let us know and it can probably be added for v13 (probably looking at May or later though..).

If it's not a big change we could try squeeze it into v12. Otherwise it won't be added into edge until we're preparing for a v13 release.

**Reply by benniekiss** [reply]
So I tested by creating `/dms/config/dovecot/10-rootless.conf` with 
```
#disable chroot for dovecot 
service anvil {
  chroot = 
}
service imap-login {
  chroot = 
}
service pop3-login {
  chroot = 
}
```

and I mapped that to `/etc/dovecot/conf.d/10-rootless.conf` in the container 
`-v /dms/config/dovecot/10-rootless.conf:/etc/dovecot/conf.d/10-rootless.conf:Z`

restarted the container, and I can login!

My whole system is a VM running Fedora CoreOS. I use podman as a regular non-root user to run the container. I did not need to change selinux rules or add additional permissions to the container in order to disable chroot in dovecot

EDIT: By the way, I tested running podman as root before adding `10-rootless.conf`, and I experienced the same issue, so it did not seem to be an issue with running podman as a regular user.

EDIT 2: And turns out the high cpu usage was from the postsrsd restart loop, but I see that's already being addressed!

**Reply by benniekiss** [reply]
So I found more information about why the problem only just began happening -- there was an update to podman which removed the default `CAP_CHROOT` capability, which would explain why the container had worked before.
https://github.com/containers/podman/releases/tag/v4.4.1

**Reply by polarathene** [reply]
Awesome thank you! We'll try squeeze that into v12 then! :heart: 

Opened issue: https://github.com/docker-mailserver/docker-mailserver/issues/3174
